### PR TITLE
feat(ButtonHyperLink): add link role

### PR DESCRIPTION
**Pull request by jiwang7**
# IMPORTANT

**Currently, this project is closed to any external contributions. Any pull request made against this project from external sources will likely be closed. If you would like to make changes to this project, please fork this project.**

# Guide

**This "Help" section can be deleted before submitting this pull request.**

*Update the name of this pull request to reflect the following shape:*

```
{type}/{scope?}/{message}
```

* **type** - A [conventional commit type](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional#type-enum) **REQUIRED**
* **scope** - The kabab-case scope of the changes in this request
* **message** - A short, kebab-case statement describing the changes **REQUIRED**

*Provide a general summary of the scope of the changes in this pull request.*

* [Readme](https://github.com/momentum-design/momentum-react-v2/blob/master/README.md)
* [Getting Started Guide](https://github.com/momentum-design/momentum-react-v2/blob/master/GETTING_STARTED.md)
* [Contributing Guide](https://github.com/momentum-design/momentum-react-v2/blob/master/CONTRIBUTING.md)

# Description

- Add role for ButtonHyperLink component to allow it to SR as a link;
- Add package —— @react-aria/link
- update package version —— @react-aria/utils 3.24.1

*The full description of the changes made in this request.*

# Links
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-533449
*Links to relevent resources.*
src/components/ButtonHyperlink

### CODE REVIEW

File: src/components/ButtonHyperlink/ButtonHyperlink.constants.ts [hunk 1]
```diff
@@ -10,4 +10,6 @@ const STYLE = {
   wrapper: `${CLASS_PREFIX}-wrapper`,
 };
 
+export type roleType = 'button' | 'link';
```
Comment (maxinteger) [inline src/components/ButtonHyperlink/ButtonHyperlink.constants.ts:13]: this must be defined in the `ButtonHyperlink.types.ts`

File: src/components/ButtonHyperlink/ButtonHyperlink.types.ts [hunk 2]
```diff
@@ -1,7 +1,10 @@
 import { ReactNode } from 'react';
 import { AriaButtonProps } from '@react-types/button';
+import { AriaLinkProps } from '@react-types/link';
+import { roleType } from './ButtonHyperlink.constants';
 
-export interface Props extends AriaButtonProps<'a'> {
+type myProps = AriaLinkProps & AriaButtonProps<'a'>;
+export interface Props extends myProps {
```
Comment (maxinteger) [inline src/components/ButtonHyperlink/ButtonHyperlink.types.ts:7]: you can extends multiple interfaces, no need for the extra type

```suggestion
export interface Props extends AiaLinkProps, AriaButtonProps<'a'> {
```

**Review by maxinteger — CHANGES_REQUESTED** [review]

File: src/components/ButtonHyperlink/ButtonHyperlink.constants.ts (hunk 1 above)
Comment (jiwang7) [reply src/components/ButtonHyperlink/ButtonHyperlink.constants.ts]: fixed

**Comment by jiwang7** [comment]
This draft change mix a button and a link. I will create another component to handle link button. Thanks for your review～

**Pull request closed without merging**